facebookresearch/multimodal
# [Lint] Upgrade usort to 1.0.2; add flake8 plugins

Summary:
- Upgrade `usort` to align with Fbcode
- The following plugins are added to `flake8`:
    - [`flake8-bugbear`](https://github.com/PyCQA/flake8-bugbear) to align with fbcode flake8 on detecting bugs/design issues
    - [`pep8-naming`](https://github.com/PyCQA/pep8-naming): naming convention checker
- Consolidate lint/formater installation into `requirements-lint.txt`.

Test Plan:
- Verify that the plugins are installed:
```
pip install -r requirements-lint.txt
flake8 --version
4.0.1 (flake8-bugbear: 22.4.25, mccabe: 0.6.1, naming: 0.12.1, pycodestyle: 2.8.0, pyflakes: 2.4.0)
```

- Create a test file to invoke the corresponding errors:
```
test_flake8.py:4:6: N802 function name 'Incorrect_Function_Name' should be lowercase
test_flake8.py:9:9: B015 Pointless comparison. This comparison does nothing but waste CPU instructions. Either prepend `assert` or remove it.
test_flake8.py:10:5: B001 Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on.  Prefer `except Exception:`.  If you're sure what you're doing, be explicit and write `except BaseException:`.
test_flake8.py:10:5: E722 do not use bare 'except'```
```
